[FOR TESTING] Enable Untracked Cache, Disable FS Monitor, Disable FS Cache

jeffhostetler opened the pull request:
One more commit than #382: disable FSCache feature.  Checking for failures in the Scalar functional tests.

derrickstolee committed:
FOR TESTING: only run scalar tests on Windows

derrickstolee committed:
FOR TESTING: enable untracked cache unconditionally

derrickstolee committed:
FOR TESTING: disable fs monitor

Signed-off-by: Derrick Stolee <dstolee@microsoft.com>

jeffhostetler committed:
FOR TESTING: disable FSCache

Signed-off-by: Jeff Hostetler <jeffhost@microsoft.com>

jeffhostetler commented:
Darn.  Looks like it also failed.

dscho commented on .github/workflows/scalar-functional-tests.yml:
```diff
@@ -20,7 +20,7 @@ jobs:
       matrix:
         # Order by runtime (in descending order)
-        os: [windows-2019, macos-10.15, ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        os: [windows-2019]
```
We should restrict this even further. In my tests, running _just_ `--test=GitCommandsTests(SparseMode).AddFileAndCommitOnNewBranchSwitchDeleteFolderAndSwitchBack()` already triggered the error, and this could potentially speed up our edit/push/analyze cycle.

derrickstolee replied:
This is a consistent failure? Ok.

1. Create a new branch in `microsoft/scalar` (or your fork of it)
2. Add `[Category("BROKEN")]` to this test.
3. In `Scalar.FunctionalTests/Program.cs` change this line:

```
            HashSet<string> includeCategories = new HashSet<string>() { "BROKEN" };
```

4. Point your `microsoft/git` branch to clone that Scalar branch for the functional tests.

dscho replied:
Nice!

dscho committed:
TO-DROP: skip CI builds while debugging the untracked cache

We are not at all interested in Git's test suite for the moment.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
TO-DROP: restrict tests even further

In my previous tests, this was sufficient to trigger the problem. In the
interest of reducing the edit/push/build cycle time, let's zero in on
the problematic part of the Functional Tests.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
TO-DROP: skip whitespace check

Leave me alone, whitespace check. We are investigating, not polishing
code.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho committed:
fixup! TO-DROP: restrict tests even further

dscho commented:
I guess we could remove/comment out the other test cases in `GitCommandsTests` to accelerate even further.

derrickstolee commented:
> I guess we could remove/comment out the other test cases in `GitCommandsTests` to accelerate even further.

Please [see my comment about how to do this without deleting tests](https://github.com/microsoft/git/pull/383#discussion_r656209688).

jeffhostetler commented:
ancient. abandoning.
